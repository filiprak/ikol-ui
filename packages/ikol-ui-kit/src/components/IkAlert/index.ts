import IkAlert from './IkAlert.vue';
export type IkAlert = InstanceType<typeof IkAlert>;
export { IkAlert };
