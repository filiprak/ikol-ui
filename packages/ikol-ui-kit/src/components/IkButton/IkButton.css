.ik-button {
    display: inline-block;
    overflow: hidden;
    text-overflow: ellipsis;
    padding: 4px 16px;
    font-size: 13px;
    font-weight: 600;
    line-height: 24px;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;

    -ms-touch-action: manipulation;
    touch-action: manipulation;
    cursor: pointer;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;

    border-radius: 8px;
}

.ik-button:hover {
    opacity: .85;
}

.ik-button--block {
    display: block;
    width: 100%;
}

/* fix black chrome outline */
.ik-button,
.ik-button:focus,
.ik-button:hover,
.ik-button:active {
    outline: none !important;
}

.ik-button:active {
    transform: translateY(1px);
}

.ik-button .ik-loader-circle {
    margin-bottom: 2px;
    margin-right: 7px;
}

.ik-button--round {
    border-radius: 48px;
}

.ik-button--size-lg {
    height: 48px;
    padding: 12px 24px;
}

.ik-button--fab {
    position: absolute;
    right: 0;
    bottom: 0;
    width: 56px;
    height: 56px;
    padding: 16px;
    border-radius: 28px;
}

.ik-button--fab.ik-button--filled.ik-button--primary {
    box-shadow: 0 5px 15px 0 var(--primary-color-o1);
}

.ik-button--fab.ik-button--filled.ik-button--success {
    box-shadow: 0 5px 15px 0 var(--success-color-o1);
}

.ik-button--fab.ik-button--filled.ik-button--error {
    box-shadow: 0 5px 15px 0 var(--error-color-o1);
}

/* default buttons */
.ik-button--filled {
    background-color: var(--surface-darken-color);
    color: var(--on-surface-darken-color-3);
}

.ik-button--flat,
.ik-button--outline {
    color: var(--on-surface-darken-color-3);
}

.ik-button--flat:hover,
.ik-button--flat:focus,
.ik-button--flat.ik-button--active,
.ik-button--flat:active,
.ik-button--outline:hover,
.ik-button--outline:focus,
.ik-button--outline.ik-button--active,
.ik-button--outline:active {
    background-color: var(--surface-darken-color);
    color: var(--on-surface-darken-color-3);
}

.ik-button--outline,
.ik-button--outline:hover,
.ik-button--outline:active,
.ik-button--outline:focus {
    border: 1px solid var(--border-color-1);
}

.ik-button.ik-button--fab {
    box-shadow: 0 5px 15px 0 var(--surface-darken-color);
}

/* primary buttons */
.ik-button--primary.ik-button--filled {
    background-color: var(--primary-color);
    color: var(--on-primary-color);
}

.ik-button--primary.ik-button--outline,
.ik-button--primary.ik-button--flat {
    color: var(--primary-color);
}

.ik-button--primary.ik-button--outline:hover,
.ik-button--primary.ik-button--outline:focus,
.ik-button--primary.ik-button--outline.ik-button--active,
.ik-button--primary.ik-button--outline:active,
.ik-button--primary.ik-button--flat:hover,
.ik-button--primary.ik-button--flat:focus,
.ik-button--primary.ik-button--flat.ik-button--active,
.ik-button--primary.ik-button--flat:active {
    background-color: var(--primary-color-o1);
    color: var(--primary-color);
}

.ik-button--primary.ik-button--outline,
.ik-button--primary.ik-button--outline:hover,
.ik-button--primary.ik-button--outline:active,
.ik-button--primary.ik-button--outline:focus {
    border: 1px solid var(--primary-color);
}

/* success buttons */
.ik-button--success.ik-button--filled,
.ik-button--success.ik-button--filled.ik-button--active,
.ik-button--success.ik-button--filled:focus,
.ik-button--success.ik-button--filled:hover,
.ik-button--success.ik-button--filled:active {
    background-color: var(--success-color);
    color: var(--on-success-color);
}

.ik-button--success.ik-button--outline,
.ik-button--success.ik-button--flat {
    color: var(--success-color);
}

.ik-button--success.ik-button--outline:hover,
.ik-button--success.ik-button--outline:focus,
.ik-button--success.ik-button--outline.ik-button--active,
.ik-button--success.ik-button--outline:active,
.ik-button--success.ik-button--flat:hover,
.ik-button--success.ik-button--flat:focus,
.ik-button--success.ik-button--flat.ik-button--active,
.ik-button--success.ik-button--flat:active {
    background-color: var(--success-color-o1);
    color: var(--success-color);
}

.ik-button--success.ik-button--outline,
.ik-button--success.ik-button--outline:hover,
.ik-button--success.ik-button--outline:active,
.ik-button--success.ik-button--outline:focus {
    border: 1px solid var(--success-color);
}

/* error buttons */
.ik-button--error.ik-button--filled,
.ik-button--error.ik-button--filled.ik-button--active,
.ik-button--error.ik-button--filled:focus,
.ik-button--error.ik-button--filled:hover,
.ik-button--error.ik-button--filled:active {
    background-color: var(--error-color);
    color: var(--on-error-color);
}

.ik-button--error.ik-button--outline,
.ik-button--error.ik-button--flat {
    color: var(--error-color);
}

.ik-button--error.ik-button--outline:hover,
.ik-button--error.ik-button--outline:focus,
.ik-button--error.ik-button--outline.ik-button--active,
.ik-button--error.ik-button--outline:active,
.ik-button--error.ik-button--flat:hover,
.ik-button--error.ik-button--flat:focus,
.ik-button--error.ik-button--flat.ik-button--active,
.ik-button--error.ik-button--flat:active {
    background-color: var(--error-color-o1);
    color: var(--error-color);
}

.ik-button--error.ik-button--outline,
.ik-button--error.ik-button--outline:hover,
.ik-button--error.ik-button--outline:active,
.ik-button--error.ik-button--outline:focus {
    border: 1px solid var(--error-color) !important;
}

/* accent buttons */
.ik-button--accent.ik-button--filled,
.ik-button--accent.ik-button--filled.ik-button--active,
.ik-button--accent.ik-button--filled:focus,
.ik-button--accent.ik-button--filled:hover,
.ik-button--accent.ik-button--filled:active {
    background-color: var(--accent-color-o1);
    color: var(--accent-color);
}

.ik-button--accent.ik-button--outline,
.ik-button--accent.ik-button--flat {
    color: var(--accent-color);
}

.ik-button--accent.ik-button--outline:hover,
.ik-button--accent.ik-button--outline:focus,
.ik-button--accent.ik-button--outline.ik-button--active,
.ik-button--accent.ik-button--outline:active,
.ik-button--accent.ik-button--flat:hover,
.ik-button--accent.ik-button--flat:focus,
.ik-button--accent.ik-button--flat.ik-button--active,
.ik-button--accent.ik-button--flat:active {
    background-color: var(--accent-color-o1);
    color: var(--accent-color);
}

.ik-button--accent.ik-button--outline,
.ik-button--accent.ik-button--outline:hover,
.ik-button--accent.ik-button--outline:active,
.ik-button--accent.ik-button--outline:focus {
    border: 1px solid var(--accent-color);
}

/* common button states */
.ik-button--circle {
    height: 32px;
    width: 32px;
    padding: 4px;
    border-radius: 50%;
}

.ik-button--circle.ik-button--size-lg {
    width: 48px;
    height: 48px;
    padding: 12px;
}

/* skeleton */
.ik-button--skeleton,
.ik-button--skeleton:active,
.ik-button--skeleton:focus {
    pointer-events: none;
    background: var(--skeleton-color-1) !important;
    border-color: var(--skeleton-color-1) !important;
    color: transparent !important;
}

.ik-button--skeleton > *,
.ik-button--skeleton:focus > *,
.ik-button--skeleton:active > * {
    visibility: hidden;
}

/* separated buttons */
.ik-button--separate {
    padding: 0 10px 0 0;
    background: none;
    background-color: transparent;
    border: none;
    color: var(--on-surface-darken-color-3);
}

.ik-button--separate .ik-icon {
    width: 32px;
    height: 32px;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    border-radius: 50%;
    background-color: var(--surface-darken-color);
}

.ik-button--separate.ik-button--size-lg .ik-icon {
    width: 48px;
    height: 48px;
}

.ik-button--separate.ik-button--primary {
    color: var(--primary-color);
}

.ik-button--separate.ik-button--primary .ik-icon {
    background-color: var(--primary-color-o1);
}

.ik-button--separate.ik-button--success {
    color: var(--success-color);
}

.ik-button--separate.ik-button--success .ik-icon {
    background-color: var(--success-color-o1);
}

.ik-button--separate.ik-button--error {
    color: var(--error-color);
}

.ik-button--separate.ik-button--error .ik-icon {
    background-color: var(--error-color-o1);
}

.ik-button--separate.ik-button--accent {
    color: var(--accent-color);
}

.ik-button--separate.ik-button--accent .ik-icon {
    background-color: var(--accent-color-o1);
}

.ik-button--loading,
.ik-button--disabled {
    opacity: 0.65;
    pointer-events: none;
}
