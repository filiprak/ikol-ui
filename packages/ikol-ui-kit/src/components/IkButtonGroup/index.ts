import IkButtonGroup from './IkButtonGroup.vue';
export type IkButtonGroupT = InstanceType<typeof IkButtonGroup>;
export { IkButtonGroup };
