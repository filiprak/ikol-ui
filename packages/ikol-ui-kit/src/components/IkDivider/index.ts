import IkDivider from './IkDivider.vue';
export { IkDivider };
