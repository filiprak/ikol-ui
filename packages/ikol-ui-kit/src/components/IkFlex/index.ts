import IkFlex from './IkFlex.vue';
export type IkFlex = InstanceType<typeof IkFlex>;
export { IkFlex };
