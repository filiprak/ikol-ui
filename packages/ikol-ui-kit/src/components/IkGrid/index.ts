import IkGrid from './IkGrid.vue';
import IkGridItem from './IkGridItem.vue';
export type IkGridT = InstanceType<typeof IkGrid>;
export type IkGridItemT = InstanceType<typeof IkGridItem>;
export { IkGrid, IkGridItem };
