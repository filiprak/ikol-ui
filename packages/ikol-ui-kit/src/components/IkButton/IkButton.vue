<template>
    <component :is="tag"
               :type="type"
               :class="button_class"
               :disabled="disabled || loading">
        <template v-if="loading">
            <IkLoaderCircle indeterminate
                            :size="16"
                            :thickness="2" />
            <template v-if="!fab && $slots.default">
                Loading...
            </template>
        </template>
        <template v-else>
            <IkIcon v-if="prepend_icon || icon"
                    :icon="prepend_icon || icon"
                    :class="prepend_icon_class" />
            <slot></slot>
            <IkIcon v-if="append_icon"
                    :icon="append_icon"
                    :class="append_icon_class" />
        </template>
    </component>
</template>
<script lang="ts">
import '@/styles';
import './IkButton.css';
import { IkIcon } from '@/components/IkIcon';
import { IkLoaderCircle } from '@/components/IkLoaderCircle';
import { computed, defineComponent } from 'vue';

export default defineComponent({
    name: 'IkButton',
    components: {
        IkIcon,
        IkLoaderCircle,
    },
    props: {
        tag: {
            type: String,
            default: 'button',
        },
        type: {
            type: String,
            default: 'button',
        },
        block: {
            type: Boolean,
            default: false,
        },
        design: {
            type: String,
            default: 'default',
        },
        size: {
            type: String,
            default: null,
        },
        prepend_icon: {
            type: String,
            default: null,
        },
        append_icon: {
            type: String,
            default: null,
        },
        icon: {
            type: String,
            default: null,
        },
        separate: {
            type: Boolean,
            default: false,
        },
        loading: {
            type: Boolean,
            default: false,
        },
        skeleton: {
            type: Boolean,
            default: false,
        },
        disabled: {
            type: Boolean,
            default: false,
        },
        fab: {
            type: Boolean,
            default: false,
        },
        round: {
            type: Boolean,
            default: false,
        },
        circle: {
            type: Boolean,
            default: false,
        },
        active: {
            type: Boolean,
            default: false,
        },
        variant: {
            type: String,
            default: null,
        },
        flat: {
            type: Boolean,
            default: false,
        },
        filled: {
            type: Boolean,
            default: false,
        },
        outline: {
            type: Boolean,
            default: false,
        },
    },
    setup(props, { slots }) {
        return {
            button_class: computed(() => {
                let variant: string | null = 'filled';

                if (props.variant) {
                    variant = props.variant;
                } else if (props.filled) {
                    variant = 'filled';
                } else if (props.outline) {
                    variant = 'outline';
                } else if (props.flat) {
                    variant = 'flat';
                }

                if (props.separate) {
                    variant = null;
                }

                const classes: { [k: string]: boolean } = {
                    'ik-button': true,
                    'ik-button--separate': props.separate,
                    'ik-button--disabled': props.disabled,
                    'ik-button--loading': props.loading,
                    'ik-button--round': props.round,
                    ['ik-button--' + props.design]: !!props.design,
                    ['ik-button--size-' + props.size]: !!props.size,
                    'ik-button--skeleton': props.skeleton,
                    'ik-bg-skeleton': props.skeleton,
                    'ik-button--block': props.block,
                    'ik-button--fab': props.fab,
                    'ik-button--circle': props.circle,
                    ['ik-button--' + variant]: !!variant,
                    'ik-button--active': props.active,
                };
                return classes;
            }),
            prepend_icon_class: computed(() => {
                return {
                    'ik-mr-2': !!slots.default?.(),
                };
            }),
            append_icon_class: computed(() => {
                return {
                    'ik-ml-2': !!slots.default?.(),
                };
            }),
        };
    },
});
</script>
