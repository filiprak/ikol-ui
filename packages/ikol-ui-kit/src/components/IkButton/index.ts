import IkButton from './IkButton.vue';
export { IkButton };
