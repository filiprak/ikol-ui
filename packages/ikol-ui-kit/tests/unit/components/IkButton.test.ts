import { mount } from '@vue/test-utils';
import { IkButton } from '@/components/IkButton';

describe('IkButton', () => {
    const getInstance = (options = {}) => mount(IkButton, options);

    it('renders with correct default state', () => {
        const wrapper = getInstance();
        expect(wrapper.classes()).toContain('ik-button');
        expect(wrapper.classes()).toContain('ik-button--default');
        expect(wrapper.classes()).toContain('ik-button--filled');
        expect(wrapper.element.tagName).toBe('BUTTON');
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders with design', () => {
        const wrapper = getInstance({
            props: {
                design: 'primary',
            },
        });
        expect(wrapper.classes()).toContain('ik-button--primary');
    });

});
