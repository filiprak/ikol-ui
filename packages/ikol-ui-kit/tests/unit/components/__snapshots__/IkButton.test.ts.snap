// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IkButton renders with correct default state 1`] = `
<button
  class="ik-button ik-button--default ik-button--filled"
  type="button"
>
  
  <!--v-if-->
  
  
  <!--v-if-->
  
</button>
`;
